osu.Game/Updater/UpdateManager.cs

```diff
@@ -44,6 +55,26 @@ protected override void LoadComplete()
             config.Set(OsuSetting.Version, version);
         }
 
+        private readonly object updateTaskLock = new object();
+
+        private Task updateCheckTask;
+
+        public async Task CheckForUpdateAsync()
+        {
+            if (!CanCheckForUpdate)
+                return;
+
+            lock (updateTaskLock)
+                updateCheckTask ??= PerformUpdateCheck();
+
+            await updateCheckTask;
```

Comment:
Safest would be to store the above to a local variable and await on the local variable here. T1 could be executing the statement above, and T2 could be blocked on the statement below, after which T2 will unblock and set null and T1 will try `await null;`